bin/cli.js

```diff
@@ -134,9 +143,28 @@ const setRequestHeaders = () => {
   addUserAgentHeader('HubSpot CLI', pkg.version);
 };
 
+const loadConfigMiddleware = async options => {
+  if (configFileExists(true)) {
+    loadConfig('', options);
+  }
+
+  if (options.config) {
```

Comment:
We discussed changing this to load the config for all commands apart from `init`, but the tests were breaking with that change. Going to do some tests with/without the `--config` flag to see if this breaks anything.

Reply:
Did the tests go well on this?  Or did this need to be updated

Reply:
You could put both these if statements in the same block if you wanted to (i.e. no need to nest).

It might be useful to add a comment in here to explain that we need to load the config when `options.config` exists so that `getAccountIdFromConfig()` in `injectAccountIdMiddleware` reads from the right config.

It might also be useful to prevent users from trying to use `--config` while on the new, hidden config since they're incompatible with each other. I think we'd want to prevent both of these scenarios:
- `configFileExists(true) === true && options.config` (i.e. new config exists, but the user is trying to override the config path)
- `options.useHiddenConfig && options.config` (i.e. the user is opting into new config, but also trying to override the default location)

Reply:
@joe-yeager, the tests went well, when I tested out commands with the `--config` flag. I did not discover any bugs, but I would appreciate if you or other members of the team could continue to test things out on the experimental branch. 

@brandenrodgers, I can add in those comments. I can also add the check for `configFileExists(true) === true && options.config`. We already check for `useHiddenConfig` and `config` with this [line](https://github.com/HubSpot/hubspot-cli/blob/44d4e2eee12276783dae40ce8e44a9d4fa8b4d03/commands/init.ts#L230).

Reply:
Addressed in https://github.com/HubSpot/hubspot-cli/pull/1123/commits/4d29cfa5d53a3cd02049061caea52a460d5ef11f